windows-containers/productization.md

```diff
@@ -0,0 +1,180 @@
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+
+The actions that the WMCB will perform are:
+* Install and configure kubelet
+* Parse the worker ignition and extract the bootstrap kubeconfig and the
+kubelet configuration
+  * This is the only use case for the ignition file
+* Launch the kubelet as a Windows service
+* Check if the kubelet is running
+* Exit
+
+### Risks and Mitigations
+
+The main risk with this proposal is the dependency on Microsoft to publish a
+downstream version of the kubelet. If this does not happen we would have to
+use the upstream version of the kubelet which will result in Red Hat being
+responsible for security and other fixes.
+
+## Design Details
+
+### Test Plan
+
+We plan to have all the repositories associated with this effort fully
+integrated with Prow CI and run e2e tests for every PR that is opened. These
+e2e tests will involve bringing up a cluster on all supported cloud providers,
+instantiating a Windows node and running workloads on it. We also plan to add
+blocking tests to the nightly runs.
+
+### Graduation Criteria
+
+This enhancement will start as GA
+
+### Upgrade / Downgrade Strategy
+
+We will support upgrades of the node components by publishing a new version
+of the WSU Ansible playbook. An older version of the playbook can be used to
+downgrade.
+
+### Version Skew Strategy
+
+We plan to maintain kubelet major version parity with the Linux counterpart.
+
+## Implementation History
+
+v1: Initial proposal
+
+## Drawbacks
+
+The Microsoft container ecosystem is not fully mature or on par with Linux
+containers. For one we can only support Windows process containers that
+restricts workloads that are runnable on Windows Server 2019 as there is a tight
+kernel version coupling with Windows containers. The coupling is slightly
+loosened if we use Hyper-V containers but that is not a viable option yet with
+all cloud providers as it requires nested virtualization support. Given these
+limitations customers could potentially get a degraded experience with Windows
+workloads when compared to Linux workloads on the cluster.
+
+## Alternatives
+
+An alternate approach to this design is to follow the operator approach where in
+the user will first install an operator from OperatorHub. The operator will
+introduce a CRD that will take similar inputs that the Ansible playbook does.
+It will then ensure that the WMCB is installed and launched on the cluster. The
+WMCB could also potentially become Windows Machine Config Daemon (WMCD) that
+supports a run once option. While it is a daemon it would run as a Windows
+service constantly reconciling with a config object on the cluster to ensure the
+node is in the desired state.
+
+We opted not to go with this approach given then time frame all the unknowns
+present in this project.
+
+## Infrastructure Needed
+
+We plan to house the WSU Ansible playbook in the openshift/release repo. We will
```

Comment:
I think this should be in `openshift-ansible` repo. https://github.com/openshift/openshift-ansible

Reply:
Sure. You said `release` repo last time.